novawallet/Modules/NetworkDetails/NetworkDetailsPresenter.swift

```diff
@@ -0,0 +1,230 @@
+import SoraFoundation
+
+final class NetworkDetailsPresenter {
+    weak var view: NetworkDetailsViewProtocol?
+    let wireframe: NetworkDetailsWireframeProtocol
+    let interactor: NetworkDetailsInteractorInputProtocol
+
+    private var chain: ChainModel
+    private var sortedNodes = SortedNodes()
+    private var connectionStates: [String: ConnectionState] = [:]
+    private var nodes: [String: ChainNodeModel] = [:]
+    private var nodesIndexes: [String: IndexPath] = [:]
+    private var selectedNode: ChainNodeModel?
+
+    private let viewModelFactory: NetworkDetailsViewModelFactory
+
+    init(
+        interactor: NetworkDetailsInteractorInputProtocol,
+        wireframe: NetworkDetailsWireframeProtocol,
+        chain: ChainModel,
+        viewModelFactory: NetworkDetailsViewModelFactory
+    ) {
+        self.interactor = interactor
+        self.wireframe = wireframe
+        self.chain = chain
+        self.viewModelFactory = viewModelFactory
+    }
+}
+
+// MARK: NetworkDetailsPresenterProtocol
+
+extension NetworkDetailsPresenter: NetworkDetailsPresenterProtocol {
+    func setup() {
+        interactor.setup()
+    }
+
+    func setNetwork(enabled: Bool) {
+        interactor.setSetNetworkConnection(enabled: enabled)
+    }
+
+    func setAutoBalance(enabled: Bool) {
+        interactor.setAutoBalance(enabled: enabled)
+    }
+
+    func addNode() {
+        wireframe.showAddNode(
+            from: view,
+            chainId: chain.chainId
+        )
+    }
+
+    func selectNode(at indexPath: IndexPath) {
```

Comment:
I really don't like that we are exposing index path and section based layout to the presenter. It is not making the UI flexible and harder to change it in future. We could just receive here a view model and find the corresponding model based on some id.